dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/annotations/PermissionNeeded.java

```diff
@@ -0,0 +1,28 @@
+package edu.harvard.iq.dataverse.annotations;
+
+import edu.harvard.iq.dataverse.persistence.user.Permission;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation is used to provide data for @Restricted annotation. It must be placed both inside @Restricted
+ * annotation and on affected PARAMETERS.
+ * <br>
+ * The usage is following: we place the annotation inside @Restricted annotation (possibly multiple times) to configure
+ * needed permissions. If there are more that one annotation on this level, attribute <b>on</b> should be used to
+ * differentiate between target parameters – and in this case corresponding parameters should have the same <b>value</b>
+ * on PARAMETER level annotation. If there is no <b>on</b> attribute filled it will be used for every parameter without
+ * filled <b>value</b>.
+ */
+@Target({ElementType.METHOD, ElementType.PARAMETER})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PermissionNeeded {
```

Comment:
Maybe add one or two examples of method signature (could be taken from tests). Something like:
```
Usage example:
<code>
        @Restricted({
                @PermissionNeeded(on = "source", needs = {Permission.EditDataset}),
                @PermissionNeeded(needs = {Permission.AddDataset})
        })
        public void method(@PermissionNeeded("source") DvObject o1, @PermissionNeeded DvObject o2)
</code>
```